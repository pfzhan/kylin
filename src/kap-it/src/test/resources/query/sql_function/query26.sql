--
-- Copyright (C) 2020 Kyligence Inc. All rights reserved.
--
-- http://kyligence.io
--
-- This software is the confidential and proprietary information of
-- Kyligence Inc. ("Confidential Information"). You shall not disclose
-- such Confidential Information and shall use it only in accordance
-- with the terms of the license agreement you entered into with
-- Kyligence Inc.
--
-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-- "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-- LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-- A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-- OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-- SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-- LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-- DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-- THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-- OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
--

SELECT  ORDER_ID,CASE length(substr(initcapb(LSTG_FORMAT_NAME),1,1))
                WHEN '1'     THEN 'LSTG_FORMAT_NAME length is 1'
                WHEN '2'     THEN 'LSTG_FORMAT_NAME length is 2'
                WHEN '3'     THEN 'LSTG_FORMAT_NAME length is 3'
                ELSE 'LSTG_FORMAT_NAME length is x' END
FROM    TEST_KYLIN_FACT
order by ORDER_ID