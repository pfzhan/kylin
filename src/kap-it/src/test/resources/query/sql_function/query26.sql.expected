-- https://github.com/Kyligence/KAP/issues/13613

SELECT  ORDER_ID,CASE length(substr(initcap(LSTG_FORMAT_NAME),1,1))
                WHEN '1'     THEN 'LSTG_FORMAT_NAME length is 1'
                WHEN '2'     THEN 'LSTG_FORMAT_NAME length is 2'
                WHEN '3'     THEN 'LSTG_FORMAT_NAME length is 3'
                ELSE 'LSTG_FORMAT_NAME length is x' END
FROM    TEST_KYLIN_FACT
order by ORDER_ID